python/sdist/amici/jax/petab.py

```diff
@@ -588,6 +614,38 @@ def _get_nominal_parameter_values(
                         model,
                         model_pars[net_id][layer_id][attribute],
                     )
+
+        # set inputs in the model if provided
+        if len(nn_input_arrays) > 0:
+            for net_id in model_pars:
+                for input in model.nns[net_id].inputs:
+                    input_array = dict(
+                        [
+                            (
+                                input,
+                                dict(
+                                    [
+                                        (
+                                            k,
+                                            jnp.array(
+                                                nn_input_arrays[net_id][input][
+                                                    k
+                                                ][:],
+                                                dtype=jnp.float64,
+                                            ),
+                                        )  # ?? hardcoded dtype not ideal ?? could infer from env somehow ??
```

Comment:
is it really necessary to set `dtype` here? Usually jax infers float precision from https://docs.jax.dev/en/latest/config_options.html. Might be necessary to cast this as numpy array first if conversion from hdf5 is the problem

Reply:
I tried casting to a numpy array but the array persisted as float32s. I've defined the `dtype` based on the current jax config settings which I think is better than hard coding.